scarlet2/nn.py

```diff
@@ -0,0 +1,156 @@
+# -------------------------------------------------- #
+# This class is used to calculate the gradient       #
+# of the log-probability via calling the nn prior    #
+# ScoreNet model. A custom vjp is created to return  #
+# the nn prior when calling jax.grad()               #
+# -------------------------------------------------- #
+
+try:
+    import numpyro.distributions as dist
+    import numpyro.distributions.constraints as constraints
+    from galaxygrad import HSC_ScoreNet32, HSC_ScoreNet64, ZTF_ScoreNet32, ZTF_ScoreNet64,HSC_LogNet32, HSC_LogNet64# (https://pypi.org/project/galaxygrad/0.0.19)
+
+except ImportError:
+    raise ImportError("scarlet2.nn requires numpyro and galaxygrad=0.0.19")
+
+import jax.numpy as jnp
+import jax.scipy as jsp
+from jax import custom_vjp
+from jax import vjp
+
+
+def pad_fwd(x, trained_model):
+    """Zero-pads the input image to the nearest 32 or 64 pixels"""
+    data_size = x.shape[1]
+    pad = True
+    
+    # select the HSC trained model
+    if trained_model in ('hsc', 'None'):
+        if data_size <= 32:
+            pad_gap = 32 - data_size
+            ScoreNet = HSC_ScoreNet32
+        else:
+            pad_gap = 64 - data_size
+            ScoreNet = HSC_ScoreNet64
+
+    
+    # select the ZTF trained model
+    elif trained_model == 'ztf':
+        if data_size <= 32:
+            pad_gap = 32 - data_size
+            ScoreNet = ZTF_ScoreNet32
+        else:
+            pad_gap = 64 - data_size
+            ScoreNet = ZTF_ScoreNet64
+    # select the custom trained model
+    else:
+        ScoreNet = trained_model
```

Comment:
Without the model selection, this is much easier: if the shape is smaller than the max_shape, pad it.

Reply:
removed these redundancies, score model now specified by user at runtime.